src/sinks/honeycomb.rs

```diff
@@ -65,6 +71,34 @@ fn default_endpoint() -> String {
     "https://api.honeycomb.io/1/batch".to_string()
 }
 
+/// Configures how the sample rate is determined for each event. If an integer sample rate is found on the
+/// event from the configured source, it will be removed from the event body and used as the sample rate for
+/// that event for Honeycomb to inflate query results. If the sample rate source is disabled, no fields will
+/// be removed and no sample rate will be sent to Honeycomb.
+#[configurable_component]
+#[derive(Clone, Debug, Default)]
+pub enum SampleRateSource {
+    /// Use the sample rate that is added to the event by the `Sample` transform.
+    #[configurable(metadata(derived))]
+    #[default]
+    SampleTransformField,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏My gut reaction is that we'd want to avoid the breaking change for users of this sink and have the default be "Disabled" (or with my other suggestion, the serde default of `None`). But I'm not sure how impactful that will actually be to users of this sink.